x/derivatives/keeper/market_test.go

```diff
@@ -1,12 +1,87 @@
 package keeper_test
 
-// import (
-// 	"github.com/UnUniFi/chain/x/derivatives/types"
+import (
+	"time"
 
-// 	sdk "github.com/cosmos/cosmos-sdk/types"
-// )
+	sdk "github.com/cosmos/cosmos-sdk/types"
 
-// TODO: add test for
-// GetPairRate(ctx sdk.Context, pair types.Market) (*sdk.Dec, error) {
-// GetAssetPrice(ctx sdk.Context, denom string) (*pftypes.CurrentPrice, error) {
-// GetPrice(ctx sdk.Context, lhsTicker string, rhsTicker string) (pftypes.CurrentPrice, error) {
+	ununifitypes "github.com/UnUniFi/chain/types"
+	"github.com/UnUniFi/chain/x/derivatives/types"
+	pricefeedtypes "github.com/UnUniFi/chain/x/pricefeed/types"
+)
+
+func (suite *KeeperTestSuite) TestGetPairRate() {
+	// get uninitialized value
+	rate, err := suite.keeper.GetPairRate(suite.ctx, types.Market{
+		BaseDenom:  "uatom",
+		QuoteDenom: "uusdc",
+	})
+	suite.Require().Error(err)
+
+	// initialize price keeper
+	_, err = suite.app.PricefeedKeeper.SetPrice(suite.ctx, sdk.AccAddress{}, "uatom:uusdc", sdk.NewDec(12), suite.ctx.BlockTime().Add(time.Hour*3))
+	suite.Require().NoError(err)
+	params := suite.app.PricefeedKeeper.GetParams(suite.ctx)
+	params.Markets = []pricefeedtypes.Market{
+		{MarketId: "uatom:uusdc", BaseAsset: "uatom", QuoteAsset: "uusdc", Oracles: []ununifitypes.StringAccAddress{}, Active: true},
+	}
+	suite.app.PricefeedKeeper.SetParams(suite.ctx, params)
+	err = suite.app.PricefeedKeeper.SetCurrentPrices(suite.ctx, "uatom:uusdc")
+	suite.Require().NoError(err)
+
+	// get initialized value
+	rate, err = suite.keeper.GetPairRate(suite.ctx, types.Market{
```

Comment:
same
https://github.com/UnUniFi/chain/pull/403/files#r1134873957

Reply:
It used to be fine when I was testing - did you try running on local and see sort of error here? (What error is it?)

Reply:
Fixed